app/graphql/types/job_application_type.rb

```diff
@@ -0,0 +1,12 @@
+module Types
+  class JobApplicationType < Types::BaseObject
+    field :id, ID, null: false
+    field :candidate_id, Integer, null: false
+    field :job_posting_id, Integer, null: false
+    field :is_active, Boolean, null: true
+    field :created_at, GraphQL::Types::ISO8601DateTime, null: false
+    field :updated_at, GraphQL::Types::ISO8601DateTime, null: false
+    field :candidate, [CandidateType], null: true
```

Comment:
Both candidate and job_posting I had the label as a reference type to start which threw me a bunch of errors and it took me a little bit to realize what was going on until I finally labeled them correctly.